atoms/inputfield.md

```diff
@@ -1,46 +1,170 @@
-# Input field
-An input field is used to input text.
+| |  |
+|-------------|------------------|
+| **Component** | Input |
+| **Type** | Molecule ([?](http://atomicdesign.bradfrost.com/chapter-2/))|
+| **Design Spec** | *pending* |
+| **Working Examples** | *pending* |
+| **Status** | `Work in progress` |
 
-![](../images/text-input.png)
+---
+
+# Input
+
+An input allows a user to enter data, usually text.
+
+![](../images/input.png)
+
+##### Contents
+
+- [Usage](#usage)
+- [Composition](#composition)
+- [Options](#options)
+- [Types](#types)
+- [States](#states)
+- [Examples in use](#examples-in-use)
+
+---
 
 ## Usage
-There are two different types of input field available: regular and filled. Regular input fields are the default choice. Both styles are functionally the same, they differ only in visual style. Regular input fields separate the label and the text field, filled fields combine them.
 
-Regular input fields are more suited to forms with multiple inputs. Filled input fields work best when used alone for a single selection, especially in compact UIs. Filled input fields also blend into a background more, so they will be less visible to the user. This can be useful for secondary actions like search or filter.
+Inputs are used wherever a user needs to input standard text information. Inputs are often used as part of forms. An input can also be used to capture information outside of a form, perhaps as a 'Filter' or 'Search' field.
+
+---
+
+## Composition
+
+![](../images/input-composition.png)
+
+An input is made up of multiple elements, some of which are optional:
+
+1. **Label, optional**
+2. **Input element, required**
+3. **Help text, optional**
```

Comment:
Question:

Should we hide the help text when displaying an error or validation message or do the messages stack?

If the e.g. error text should be below the help text, then we can add: "4. **Validation text, optional**"

Reply:
Oh yeah, of course. Thanks for catching that. The messages stack, I've updated the example.